Source/CronosPlayUnreal/Public/DefiWalletCoreActor.h

```diff
@@ -517,281 +535,236 @@ class CRONOSPLAYUNREAL_API ADefiWalletCoreActor : public AActor {
      */
     UFUNCTION(BlueprintCallable,
               meta = (DisplayName = "Erc721TotalSupply", Keywords = "Wallet"),
               Category = "CronosPlayUnreal")
-    void Erc721TotalSupply(FString contractAddress, FString &totalsupply,
-                           bool &success, FString &output_message);
+    void Erc721TotalSupply(FString contractAddress,
+                           FGeneralCronosCallDelegate Out);
 
     /**
      *  Returns a token ID at a given index of all the tokens stored by the
      * contract. Use along with totalSupply to enumerate all tokens.
      * @param contractAddress erc 721 contract address
      * @param erc721index which index
-     * @param token a token ID at a given index
-     * @param success whether succeed or not
-     * @param output_message error message, "" if succeed
+     * @param Out callback
      */
     UFUNCTION(BlueprintCallable,
               meta = (DisplayName = "Erc721TokenByIndex", Keywords = "Wallet"),
               Category = "CronosPlayUnreal")
     void Erc721TokenByIndex(FString contractAddress, FString erc721index,
-                            FString &token, bool &success,
-                            FString &output_message);
+                            FGeneralCronosCallDelegate Out);
 
     /**
      * Returns a token ID owned by owner at a given index of its token list. Use
      * along with balanceOf to enumerate all of owner's tokens.
      * @param contractAddress erc 721 contract address
      * @param erc721owner owner
      * @param erc721index which index
-     * @param token a token ID at a given index
-     * @param success whether succeed or not
-     * @param output_message error message, "" if succeed
+     * @param Out callback
      */
     UFUNCTION(BlueprintCallable,
               meta = (DisplayName = "Erc721TokenOwnerByIndex",
                       Keywords = "Wallet"),
               Category = "CronosPlayUnreal")
     void Erc721TokenOwnerByIndex(FString contractAddress, FString erc721owner,
-                                 FString erc721index, FString &token,
-                                 bool &success, FString &output_message);
+                                 FString erc721index,
+                                 FGeneralCronosCallDelegate Out);
 
     /**
      * Get erc-1155 uri
      * @param contractAddress erc1155 contract address
      * @param tokenID token ID
-     * @param uri  get uri
-     * @param success whether succeed or not
-     * @param output_message error message, "" if succeed
+     * @param Out callback
      */
     UFUNCTION(BlueprintCallable,
               meta = (DisplayName = "Erc1155Uri", Keywords = "Wallet"),
               Category = "CronosPlayUnreal")
-    void Erc1155Uri(FString contractAddress, FString tokenID, FString &uri,
-                    bool &success, FString &output_message);
+    void Erc1155Uri(FString contractAddress, FString tokenID,
+                    FGeneralCronosCallDelegate Out);
```

Comment:
Same, better off assigning a dedicated delegate. Rename it.

Reply:
ok